compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -104,6 +104,33 @@ pub fn create_informational_target_machine(sess: &Session) -> &'static mut llvm:
         .unwrap_or_else(|err| llvm_err(sess.diagnostic(), err).raise())
 }
 
+pub fn create_target_machine_args(expanded_args: &[String]) -> *mut llvm::TargetMachineCppArgs {
+    let expanded_args_cstr: Vec<CString> =
+        expanded_args.iter().map(|s| CString::new(s.as_str()).ok()).flatten().collect();
```

Comment:
```suggestion
        expanded_args.iter().map(|s| CString::new(s).ok()).flatten().collect();
```
Deref coercion should work here?

Reply:
Did not work. 
Now that I am looking more a this it seems strange that &str implements Into<Vec<u8>>, but &String does not.